### PR TITLE
feat(core): Add isStable Observable property to ApplicationRef to ind…

### PR DESCRIPTION
**Pull request by vikerman**
…icate when it's stable and unstable.

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
-

**What is the new behavior?**
There is a new isStable Observable on ApplicationRef that emits true when application is stable and false when unstable. This can be used by frameworks like Universal to detect when the application is stable.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
-

**Other information**:

### CODE REVIEW

File: modules/@angular/core/src/application_ref.ts [hunk 1]
```diff
@@ -425,6 +433,27 @@ export class ApplicationRef_ extends ApplicationRef {
 
     this._zone.onMicrotaskEmpty.subscribe(
         {next: () => { this._zone.run(() => { this.tick(); }); }});
+
+    this._zone.onStable.subscribe({
+      next: () => {
+        // Check whether there are no pending macro/micro tasks in the next tick
```
Comment (alxhub) [inline modules/@angular/core/src/application_ref.ts]: `NgZone.assertNotInAngularZone`

File: modules/@angular/core/src/application_ref.ts [hunk 2]
```diff
@@ -6,9 +6,11 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {Subject} from 'rxjs/Subject';
```
Comment (IgorMinar) [inline modules/@angular/core/src/application_ref.ts]: use EventEmitter for consistency please

Comment (IgorMinar) [inline modules/@angular/core/src/application_ref.ts]: actually, per conversation this should be just Observable

File: modules/@angular/core/src/application_ref.ts [hunk 3]
```diff
@@ -510,4 +545,6 @@ export class ApplicationRef_ extends ApplicationRef {
   get componentTypes(): Type<any>[] { return this._rootComponentTypes; }
 
   get components(): ComponentRef<any>[] { return this._rootComponents; }
+
+  get isStable(): Subject<boolean> { return this._isStable; }
```
Comment (IgorMinar) [inline modules/@angular/core/src/application_ref.ts]: return observable

File: modules/@angular/core/test/application_ref_spec.ts [hunk 4]
```diff
@@ -124,6 +125,139 @@ export function main() {
...
+            appRef.isStable.subscribe({
+              next: (stable: boolean) => {
+                if (!stable) {
+                  unstableCalled = true;
+                }
+              }
+            });
+          }
+
+          it(' should be fired after app becomes unstable', async(() => {
```
Comment (IgorMinar) [inline modules/@angular/core/test/application_ref_spec.ts]: extra space

**Review by IgorMinar — DISMISSED** [review]

File: modules/@angular/core/src/application_ref.ts (hunk 2 above)
Comment (vikerman) [inline modules/@angular/core/src/application_ref.ts]: Using Subject internally and exposing the Observable through the public property.

File: modules/@angular/core/src/application_ref.ts (hunk 3 above)
Comment (vikerman) [inline modules/@angular/core/src/application_ref.ts]: Done

File: modules/@angular/core/test/application_ref_spec.ts (hunk 4 above)
Comment (vikerman) [inline modules/@angular/core/test/application_ref_spec.ts]: Done

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**